node/nodeBlocks.go

```
@@ -0,0 +1,34 @@
+package node
+
+import (
+	"github.com/ElrondNetwork/elrond-go-core/data"
+	"github.com/ElrondNetwork/elrond-go/process"
+)
+
+func (n *Node) getBlockHeaderByHash(hash []byte) (data.HeaderHandler, error) {
+	return process.GetHeaderFromStorage(
+		n.processComponents.ShardCoordinator().SelfId(),
+		hash,
+		n.coreComponents.InternalMarshalizer(),
+		n.dataComponents.StorageService(),
+	)
+}
+
+func (n *Node) getBlockRootHash(headerHash []byte, header data.HeaderHandler) []byte {
+	blockRootHash, err := n.processComponents.ScheduledTxsExecutionHandler().GetScheduledRootHashForHeader(headerHash)
+	if err != nil {
```

Unfortunately, we have to swallow this error.

---

Perhaps compare it with some well-known errors, then `log.Warn` if necessary?